nucliadb_node/src/services/reader.rs

```diff
@@ -485,26 +483,43 @@ impl ShardReaderService {
             }
         };
 
+        let relation_request = search_request.relations;
+        let relation_reader_service = self.relation_reader.clone();
+        let relation_task = move || {
+            if !skip_relations {
+                let results = relation_reader_service.search(&relation_request.unwrap());
+                Some(results)
+            } else {
+                None
+            }
```

Comment:
This code is a bit convoluted and can be replaced by the following functional code:
```rust
search_request
   .relations
   .map(|relations| relation_reader_service.search(relation)
```
By doing that, you get rid of the not-useful-anymore `skip_relations` variable, and make the code clearer.

Reply:
Done